core/NoAccessException.php

```diff
@@ -18,4 +19,8 @@
  */
 class NoAccessException extends InvalidRequestParameterException
 {
-}
\ No newline at end of file
+    public function __construct($message)
+    {
+        parent::__construct($message, $code = 401);
+    }
```

Comment:
As this Exception is now used like this 
```
$ex = new NoAccessException(Piwik::translate('Login_LoginNotAllowedBecauseUserLoginBlocked'), 403);
```

We need to handle the code correctly:
```suggestion
    public function __construct($message, $code = 401)
    {
        parent::__construct($message, $code);
    }
```